hummingbot/connector/derivative/coinflex_perpetual/coinflex_perpetual_derivative.py

```diff
@@ -0,0 +1,1391 @@
+            else:
+                await self._update_positions()
+
+    async def _update_positions(self):
+        positions = await self._api_request(path=CONSTANTS.POSITION_INFORMATION_URL,
+                                            is_auth_required=True,
+                                            )
+        if not positions['data']:
+            return
+        for position in positions['data']:
+            trading_pair = position.get("instrumentId")
+            amount = Decimal(position.get("quantity"))
+            position_side = PositionSide.SHORT if amount < s_decimal_0 else PositionSide.LONG
+            unrealized_pnl = Decimal(position.get("positionPnl"))
+            entry_price = Decimal(position.get("entryPrice"))
+            leverage = Decimal('1')
+            pos_key = self.position_key(trading_pair, position_side)
+            if amount != 0:
+                self._account_positions[pos_key] = Position(
+                    trading_pair=await CoinflexPerpetualAPIOrderBookDataSource.convert_from_exchange_trading_pair(
+                        exchange_trading_pair=trading_pair,
+                        domain=self._domain,
+                        throttler=self._throttler,
+                        api_factory=self._api_factory,
+                    ),
+                    position_side=position_side,
+                    unrealized_pnl=unrealized_pnl,
+                    entry_price=entry_price,
+                    amount=amount,
+                    leverage=leverage
+                )
+            else:
+                if pos_key in self._account_positions:
+                    del self._account_positions[pos_key]
+
+    async def _fetch_order_status(self, tracked_order) -> Dict[str, Any]:
+        """
+        Helper function to fetch order status.
+        Returns a dictionary with the response.
+        """
+        order_params = {
+            "marketCode": await CoinflexPerpetualAPIOrderBookDataSource.convert_to_exchange_trading_pair(
+                hb_trading_pair=tracked_order.trading_pair,
+                domain=self._domain,
+                api_factory=self._api_factory,
+                throttler=self._throttler)
+        }
+
+        # If we get the exchange order id, use that, otherwise use client order id.
+        try:
+            async with timeout(self._sleep_time(1)):
+                await tracked_order.get_exchange_order_id()
+            order_params["orderId"] = tracked_order.exchange_order_id
+        except asyncio.TimeoutError:
+            order_params["clientOrderId"] = tracked_order.client_order_id
+
+        return await self._api_request(
+            method=RESTMethod.GET,
+            path=CONSTANTS.ORDER_STATUS_URL,
+            params=order_params,
+            is_auth_required=True,
+            endpoint_api_version="v2.1")
+
+    async def _update_order_status(self):
+        """
+        Calls the REST API to get order/trade updates for each in-flight order.
+        """
```

Comment:
The logic to update order status is very good.
Since the Coinflex connector has and endpoint to query for trades it would be interesting to include the HTTP trades retrieval as a backup mechanism for the trades received through websocket (this is a nice to have and the connector can be accepted without this functionality)

Reply:
See below regarding trades endpoint